addons/stock/models/stock_orderpoint.py

```diff
@@ -413,11 +413,19 @@ def _get_orderpoint_action(self):
             to_refill[(product, warehouse)] = product_qty + qty_in_progress
         to_refill = {k: v for k, v in to_refill.items() if float_compare(
             v, 0.0, precision_digits=rounding) < 0.0}
-
-        lot_stock_id_by_warehouse = self.env['stock.warehouse'].search_read([
+        # Each warehouse should have a replenish location, use this for each warehouse instead of lot stock
+        # we need view stock since its the parent of any replenish loc in the warehouse
+        view_id_by_warehouse = self.env['stock.warehouse'].search_read([
             ('id', 'in', [g[1] for g in to_refill.keys()])
-        ], ['lot_stock_id'])
-        lot_stock_id_by_warehouse = {w['id']: w['lot_stock_id'][0] for w in lot_stock_id_by_warehouse}
+        ], ['view_location_id'], load='')
+        view_id_by_warehouse = {w['id']: w['view_location_id'] for w in view_id_by_warehouse}
+        location_by_warehouse = {}
+
+        for ware_id, view_id in view_id_by_warehouse.items():
+            replenish_locs = self.env['stock.location'].search([('replenish_location', '=', True), ('location_id', 'child_of', view_id)])
```

Comment:
It's usually not recommended to use a method that interacts with the database in a loop that could be arbitrarily big, this can create performance issues. Most of the time it's possible to use the `read_group` method instead to avoid the loop.
(There is a nice video explaining that here actually: https://www.youtube.com/watch?v=5jT5eoCZH6U&ab_channel=Odoo)